composer/utils/ddp.py

```diff
@@ -141,6 +141,49 @@ def all_reduce(
                        "The mosaic trainer will automatically do this for you.")
 
 
+def broadcast(tensor: torch.Tensor, src: int = 0) -> None:
+    """Broadcasts the tensor to the whole group.
+
+    ``tensor`` must have the same number of elements in all processes participating in the collective.
+    See :meth:`torch.distributed.broadcast`.
+
+    Args:
+        tensor (torch.Tensor): Data to be sent if ``src`` is the rank of current process,
+            and tensor to be used to save received data otherwise.
+        src (int, optional): Source rank (default: ``0``)
+    """
+    if dist.is_available() and dist.is_initialized():
+        dist.broadcast(tensor, src)
+    world_size = get_world_size()
+    if world_size == 1:
+        return
+    raise RuntimeError(f"Since the world_size({world_size}) > 1, please configure DDP to use ddp.broadcast(). "
+                       "The mosaic trainer will automatically do this for you.")
```

Comment:
Will it? Do you mean the `composer` CLI?

Reply:
Yup, the trainer will configure ddp for you. The launch script sets the env variables. (If no env variables are set, then it'll default to a world size of 1). This error would only be raised if someone is using these functions outside of our trainer, or if we attempt to access DDP before it's configured.